railgun/manager/run.go

```diff
@@ -91,12 +91,18 @@ func Run(ctx context.Context, c *config.Config) error {
 		return err
 	}
 
+	isTagExists, err := kongClient.Tags.Exists(ctx)
+	if err != nil {
+		setupLog.Error(err, "cannot able to get tags from Kong Admin API client")
+	}
+
 	// configure the kong client
 	kongConfig := sendconfig.Kong{
 		URL:               c.KongAdminURL,
 		FilterTags:        c.FilterTags,
 		Concurrency:       c.Concurrency,
 		Client:            kongClient,
+		HasTagSupport:     isTagExists,
 		PluginSchemaStore: util.NewPluginSchemaStore(kongClient),
 	}
```

Comment:
This PR implements tags support in a way very similar to how KIC 1.x did, but let's use this opportunity to simplify the implementation. We could:

1. mark the `sendconfig.Kong.HasTagSupport` field as deprecated in KIC 2.x
2. make `sendconfig` expect that, if Kong doesn't support tags, then the `FilterTags` list be empty

If this sounds right, let's:

1. rename the `HasTagSupport` field to `DeprecatedHasTagSupport` and add a documentation comment like `// DeprecatedHasTagSupport is not used in KIC 2.x. If the gateway instance does not support tags, pass an empty FilterTags slice instead.`
2. rewrite the `kongClient.Tags.Exists(ctx)` check above to something like:
    ```go
    var filterTags []string
    if ok, err := kongClient.Tags.Exists(ctx); err != nil {
      setupLog.Error(err, "cannot able to get tags from Kong Admin API client")
      return err
    } else if ok {
      setupLog.Info("tag filtering enabled", "tags", c.filterTags)
      filterTags = c.FilterTags
    }
    ```
3. drop the `HasTagSupport` field from the `kongConfig` initialization in line 100 and use `filterTags` from the code snippet above
4. update the flag description by appending `This setting will be silently ignored if the Kong instance has no tags support.`

Reply:
addressed

Reply:
My previous code suggestion in point 2 was buggy :roll_eyes: sorry about that.

I've updated the suggestion and committed the change directly to your branch.